File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        for(int i = 0; i < nums.length - 1; i++){
            for(int j = 0; j < nums.length; j++){
                if(i == j)
                    continue;
                if(target == nums[i] + nums[j]){
                    res[0] = i;
                    res[1] = j;
                    return res;
                }
            }
        }
        return res;
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    // Create a map to store the complements of numbers we have seen so far
    const complementMap = {};

    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        const complement = target - num;

        // If the complement exists in the map, we have found our pair
        if (complementMap.hasOwnProperty(complement)) {
            // Return the indices of the two numbers
            return [complementMap[complement], i];
        }

        // Otherwise, store the current number and its index
        complementMap[num] = i;
    }

    // If no solution is found, return an empty array
    return [];
};
